src/appleseed/renderer/kernel/lighting/imagebasedlighting.cpp

```diff
@@ -204,9 +204,20 @@ void compute_ibl_environment_sampling(
 
         // Discard occluded samples.
         Spectrum transmission;
-        material_sampler.trace_simple(shading_context, incoming, transmission);
-        if (is_zero(transmission))
-            continue;
+        if (environment_edf.get_cast_shadows())
+        {
+            material_sampler.trace_simple(
```

Comment:
There's another `trace_simple()` call in that file that should obey the cast shadow flag, in `compute_ibl_material_sampling()`.

Make sure to re-render the reference images for your test scenes after you've done this change.